boards/arm/96b_argonkey/CMakeLists.txt

```diff
@@ -1,5 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
+if(CONFIG_PINMUX)
 zephyr_library()
 zephyr_library_sources(pinmux.c)
```

Comment:
Why not use `zephyr_library_sources_ifdef(CONFIG_PINUX pinmux.c)` ?

Reply:
Isn't it a problem to have zephyr_library() and zephyr_library_include_directories() even when no source file to compile ?

Reply:
good question @SebastianBoe can you weigh in?

Reply:
Not sure what you meant @carlescufi , but the code as-written is good practice.

We can't use zephyr_sources_ifdef because we want to modify the include paths of pinmux only.

Also, we can't create a zephyr_library and not add any sources to it.

Reply:
Thanks for the info @SebastianBoe.